tests/foreman/cli/test_host.py

```diff
@@ -1159,504 +826,156 @@ def test_positive_list_with_nested_hostgroup(self):
-        })
-        Medium.add_operating_system({
-            'id': self.host_args.medium.id,
-            'operatingsystem-id': new_os['id'],
-        })
-        Host.update({
-            'id': self.host['id'],
-            'operatingsystem-id': new_os['id'],
-        })
-        self.host = Host.info({'id': self.host['id']})
-        self.assertEqual(
-            self.host['operating-system']['operating-system'], new_os['title'])
-
-    @tier2
-    def test_positive_update_os_by_name(self):
-        """A host can be updated with a new operating system. Use
-        entities names for association
 
-        :id: bd48887f-3db3-47b0-8231-de58884efe57
 
-        :expectedresults: A host is updated and the operating system matches
+class HostUpdateTestCase(CLITestCase):
+    """Tests for updating the hosts."""
 
-        :CaseLevel: Integration
-        """
-        new_os = make_os({
-            'architectures': self.host_args.architecture.name,
-            'partition-tables': self.host[
-                'operating-system']['partition-table'],
-        })
-        Medium.add_operating_system({
-            'name': self.host_args.medium.name,
-            'operatingsystem': new_os['title'],
-        })
-        Host.update({
-            'name': self.host['name'],
-            'operatingsystem': new_os['title'],
+    def setUp(self):
+        """Create a host to reuse later"""
+        super(HostUpdateTestCase, self).setUp()
+        self.puppet_proxy = Proxy.list({
+            'search': 'url = https://{0}:9090'.format(settings.server.hostname)
+        })[0]
+        # using nailgun to create dependencies
+        self.host_args = entities.Host()
+        self.host_args.create_missing()
+        # using CLI to create host
+        self.host = make_host({
+            u'architecture-id': self.host_args.architecture.id,
+            u'domain-id': self.host_args.domain.id,
+            u'environment-id': self.host_args.environment.id,
+            u'location-id': self.host_args.location.id,
+            u'mac': self.host_args.mac,
+            u'medium-id': self.host_args.medium.id,
+            u'name': self.host_args.name,
+            u'operatingsystem-id': self.host_args.operatingsystem.id,
+            u'organization-id': self.host_args.organization.id,
+            u'partition-table-id': self.host_args.ptable.id,
+            u'puppet-proxy-id': self.puppet_proxy['id'],
+            u'root-password': self.host_args.root_pass,
         })
-        self.host = Host.info({'name': self.host['name']})
-        self.assertEqual(
-            self.host['operating-system']['operating-system'], new_os['title'])
 
-    @tier2
-    def test_positive_update_medium_by_id(self):
-        """A host can be updated with a new medium. Use entities ids for
-        association
+    @tier1
+    def test_positive_update_parameters_by_name(self):
+        """A host can be updated with a new random name. Use id to
+        access the host
```

Comment:
Please mention here all updated entities.

- new name ( you already have it) 
- new mac address 
- new domain
- new environment
- new architecture
- new os 
- new medium